string-functions: add instrunctions for LEFT() and LENGTH() (#16068)
pingcap/docs: functions-and-operators/string-functions.md

Reviewer: @takaidohigasi, thanks for your valuable contribution.  I've added some comments on this PR. PTAL.

### [`LEFT()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_left)

The `LEFT()` function returns a specified number of characters from the left side of a string.

Syntax:

```sql
LEFT(`str`, `len`)
```

- `str`: the original string to extract characters. If `str` contains a multibyte character, the function counts it as a single code point. 
- `len`: the length of characters to be returned. 
    - If `len` is equal to or less than 0, the function returns an empty string.

>     - If `len` is greater than the length of `str`, the function returns the original `str`. 

Reviewer 2: ```suggestion
    - If `len` is equal to or greater than the length of `str`, the function returns the original `str`. 
```

### [`LEFT()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_left)

The `LEFT()` function returns a specified number of characters from the left side of a string.

> 

Reviewer: ```suggestion
```

Syntax:

```sql
LEFT(`str`, `len`)
```

- `str`: the original string to extract characters. If `str` contains a multibyte character, the function counts it as a single code point. 
- `len`: the length of characters to be returned. 
    - If `len` is equal to or less than 0, the function returns an empty string.

>     - If `len` is greater than the length of `str`, the function returns the length of `str`. 

Reviewer: ```suggestion
    - If `len` is greater than the length of `str`, the function returns the original `str`. 
```

Return the leftmost number of characters as specified.

* usage: LEFT(`str`, `len`)
    * len: length of character to return
    * str: original string to extract characters
* if `len` is equal or less than 0, the function returns empty string
* function is multibyte safe.
* If any argument is `NULL`, the function returns `NULL`.

> 

Reviewer 3: ```suggestion
* If `len` is greater than the length of `str`, the function returns `str`.

```

Reviewer 3: How about adding an explanation for `SELECT LEFT('ABCED', 6);`?

Author: thank you for the suggestion!

Return the leftmost number of characters as specified.

> * usage: LEFT(`str`, `len`)
>     * len: length of character to return
>     * str: original string to extract characters

Reviewer: ```suggestion

Syntax

```sql
LEFT(`str`, `len`)
```

- `str`: the original string to extract characters.
- `len`: length of characters to be returned. 

```

Examples:

```sql
SELECT LEFT('ABCED', 3);
+------------------+
| LEFT('ABCED', 3) |
+------------------+
| ABC              |
+------------------+

SELECT LEFT('ABCED', 6);
+------------------+
| LEFT('ABCED', 6) |
+------------------+
| ABCED            |
+------------------+
```

```sql
SELECT LEFT('ABCED', 0);
+------------------+
| LEFT('ABCED', 0) |
+------------------+
|                  |
+------------------+

SELECT LEFT('ABCED', -1);
+-------------------+
| LEFT('ABCED', -1) |
+-------------------+
|                   |
+-------------------+
```

```sql
SELECT LEFT('🍣ABC', 3);
+--------------------+
| LEFT('🍣ABC', 3)     |
+--------------------+
| 🍣AB                 |
+--------------------+
```

```sql
SELECT LEFT('ABC', NULL);
+-------------------+
| LEFT('ABC', NULL) |
+-------------------+
| NULL              |
+-------------------+

SELECT LEFT(NULL, 3);
+------------------------------+
| LEFT(NULL, 3)                |
+------------------------------+
| NULL                         |
+------------------------------+
```

### [`LENGTH()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_length)

Return the length of a string in bytes.

> `LENGTH()` count a multibyte character as multi bytes while `CHAR_LENGTH()` count it as a single code point

Reviewer: ```suggestion
`LENGTH()` counts a multibyte character as multi bytes while `CHAR_LENGTH()` counts a multibyte character as a single code point.
```

> * If any argument is `NULL`, the function returns `NULL`.

Reviewer: ```suggestion
If the argument is `NULL`, the function returns `NULL`.
```

> Return the leftmost number of characters as specified.

Reviewer: ```suggestion
The `LEFT()` function returns a specified number of characters from the left side of a string.
```

Examples:

```sql
SELECT LEFT('ABCED', 3);
+------------------+
| LEFT('ABCED', 3) |
+------------------+
| ABC              |
+------------------+

SELECT LEFT('ABCED', 6);
+------------------+
| LEFT('ABCED', 6) |
+------------------+
| ABCED            |
+------------------+
```

```sql
SELECT LEFT('ABCED', 0);
+------------------+
| LEFT('ABCED', 0) |
+------------------+
|                  |
+------------------+

SELECT LEFT('ABCED', -1);
+-------------------+
| LEFT('ABCED', -1) |
+-------------------+
|                   |
+-------------------+
```

```sql
SELECT LEFT('🍣ABC', 3);
+--------------------+
| LEFT('🍣ABC', 3)     |
+--------------------+
| 🍣AB                 |
+--------------------+
```

```sql
SELECT LEFT('ABC', NULL);
+-------------------+
| LEFT('ABC', NULL) |
+-------------------+
| NULL              |
+-------------------+

SELECT LEFT(NULL, 3);
+------------------------------+
| LEFT(NULL, 3)                |
+------------------------------+
| NULL                         |
+------------------------------+
```

### [`LENGTH()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_length)

> Return the length of a string in bytes.

Reviewer: ```suggestion
The `LENGTH()` function returns the length of a string in bytes.
```

Return the leftmost number of characters as specified.

Syntax:

```sql
LEFT(`str`, `len`)
```

> - `str`: the original string to extract characters.
> - `len`: the length of characters to be returned. 
> 
> - If `len` is equal to or less than 0, the function returns an empty string.
> - If `len` is greater than the length of `str`, the function returns the length of `str`. 
> - If `str` contains a multibyte character, the function counts it as a single code point. 
> - If any argument is `NULL`, the function returns `NULL`.

Reviewer: ```suggestion
- `str`: the original string to extract characters. If `str` contains a multibyte character, the function counts it as a single code point. 
- `len`: the length of characters to be returned. 
    - If `len` is equal to or less than 0, the function returns an empty string.
    - If `len` is greater than the length of `str`, the function returns the length of `str`. 
- If any argument is `NULL`, the function returns `NULL`.
```

### [`LENGTH()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_length)

Return the length of a string in bytes.

> `LENGTH()` count a multibyte character as multi bytes while `CHAR_LENGTH()` count it as a single code point

Reviewer 4: Some things in Unicode make this quite complicated, for example flags.  However to me it looks like the text here is fully correct :smile: 

```
mysql> WITH n AS (SELECT 'abc' p UNION ALL SELECT 'Café' UNION ALL SELECT '😀' UNION ALL SELECT '🇪🇺') SELECT p, CHAR_LENGTH(p), LENGTH(p), HEX(p) FROM n;
+----------+----------------+-----------+------------------+
| p        | CHAR_LENGTH(p) | LENGTH(p) | HEX(p)           |
+----------+----------------+-----------+------------------+
| abc      |              3 |         3 | 616263           |                     
| Café     |              4 |         5 | 436166C3A9       |
| 🇪🇺         |              2 |         8 | F09F87AAF09F87BA |
| 😀         |              1 |         4 | F09F9880         |
+----------+----------------+-----------+------------------+
4 rows in set (0.01 sec)
```

Reviewer 4: Note that the linter doesn't like the manual line break:

```
[2024-01-10T07:23:47.103Z] functions-and-operators/string-functions.md: this file has manual line breaks in the following lines:
[2024-01-10T07:23:47.103Z] 
[2024-01-10T07:23:47.103Z] MANUAL LINE BREAKS: L245
```

Maybe something like this, or merge it with line 244.

```suggestion

`LENGTH()` count a multibyte character as multi bytes while `CHAR_LENGTH()` count it as a single code point
```

Author: thanks for the review!

I agree with you and collation is more difficult for some languages related to the it...
